src/Shared/CertificateGeneration/CertificateManager.cs

```diff
@@ -228,6 +232,44 @@ public X509Certificate2 CreateAspNetCoreHttpsDevelopmentCertificate(DateTimeOffs
             return certificate;
         }
 
+        internal static bool CheckDeveloperCertificateKey(X509Certificate2 candidate)
+        {
+            // Tries to use the certificate key to validate it can't access it
+            try
+            {
+                // Search for the 'active' developer certificate, being very conservative about the possibility that there
+                // might be multiple repeated certificates and multiple HTTPS certificates that have been rolled over.
+                var devCert = ListCertificates(CertificatePurpose.HTTPS, StoreName.My,StoreLocation.CurrentUser, isValid:false)
+                    .OrderByDescending(c => c.NotAfter)
+                    .FirstOrDefault(c => candidate.Thumbprint == c.Thumbprint && c.HasPrivateKey);
+
+                if (devCert == null)
+                {
+                    return false;
+                }
+
+                var rsa = devCert.GetRSAPrivateKey();
+                if (rsa == null)
+                {
+                    return false;
+                }
+
+                // Encrypting a random value is the ultimate test for a key validity.
+                // Windows and Mac OS both return HasPrivateKey = true if there is (or there has been) a private key associated
+                // with the certificate at some point.
+                var value = new byte[32];
+                RandomNumberGenerator.Fill(value);
+                rsa.Decrypt(rsa.Encrypt(value, RSAEncryptionPadding.Pkcs1), RSAEncryptionPadding.Pkcs1);
+
+                // Being able to encrypt and decrypt a payload is the strongest guarantee that the key is valid.
+                return true;
```

Comment:
@GrabYourPitchforks I believe this is good enough, isn't it?

Reply:
No issues from me. You'll want somebody more familiar with Mac to verify that this doesn't pop UI (e.g., to provide a PIN) on the developer's machine.

Reply:
It doesn't pop a UI with the keys we create, but that was a good suggestion.

Thanks for looking into this!